### PR TITLE
Docker-ize WFL

### PR DESCRIPTION
**Pull request by rexwangcc**
### Purpose

- https://broadinstitute.atlassian.net/jira/software/projects/GH/boards/530?selectedIssue=GH-709

### Changes

- Add a Dockerfile for WFL api server.
- Add a Dockerfile for WFL ui.
- Add .dockerignore files accordingly to keep the docker build context slim.
- Updates to readme files will come later once this PR is in a good shape before merge.

### Open Questions

~1. Currently the routing for WFL is defined by GAE's custom yaml and automatically handled by GAE. When we move to GKE, this will need to be handled by the NGINX layer in front of the VueJS app. The nginx.conf file will need to be carefully crafted and configured. Not sure if it should be part of this PR?~ I have added the Nginx configuration and updated the dockerfiles accordingly.

~2. Right now the api docker image is simply copying the fat JAR from `target/` into the docker Java runtime. This will require the developer/build jobs to compile the JAR in advance to build the docker image. Alternatives are:~

    1. `boot build` the jar within a `clojure` docker image, but that will require handling the git tokens within the container, which is not easy.
    2. I remember clojure/ring-server can run without being built into a JAR, we could just run the source code in a clojure runtime docker image, this will give us the benefit to mount configs such as environments.clj from GKE configmap later, but not sure how possible this approach could be, @tbl3rd might have more insights.

After a team discussion, we have decided to re-use the existing OIDC proxy to serve the static UI files. And we will keep the API image a simple one containing the fat JAR file.

 3. With the docker image being created, a problem about versioning comes up to us now. We need a way to cut off release of WFL and its images, so GKE could reference them. Are we going to SemVer it?

4. Is it more convenient for us to have a Docker-compose file defined as well for local testing?

~5. Right now the connection to CloudSQL from WFL is handled by GAE, how could we keep the current setup after moving to GKE? https://cloud.google.com/sql/docs/mysql/connect-kubernetes-engine mentions both private IPs and CloudSQL proxy image, not sure which fits into our use cases better. @hjfbynara @samanehsan~ We decided to use the "sqlproxy JAR" which we are already using to handle this.

#### API
- `docker build -t workflow-launcher-api:test .`
- `docker run -it -p 3000:3000 --rm workflow-launcher-api:test`

#### UI
- `cd ui`
- `docker build -t workflow-launcher-ui:test .`
    ```
    docker run -it --rm --name wfl-ui --hostname localhost \
        -p 80:80 \
        -v $(pwd)/ui/site.dev.conf:/etc/apache2/sites-available/site.conf \
        workflow-launcher-ui:test
    ```

### CODE REVIEW

**Comment by rhiananthony** [comment]
For 1) I tend to think that it should be part of this work.
For 2) I don't like the building the jar inside the container. Mostly because of the git credentials or the vault credentials that it would need. It seems more straightforward to compile the JAR in advance or to see if clojure can run without needing the JAR.
For 3) I def prefer Semantic versioning. Think that would need input from the rest of Hornet though.
for 4) Yes
for 5) Take a look at how cromwell is doing it. Cromwell needs to talk to CloudSQL from GKE too.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM openjdk:11
```
[the remark this answers is not in the document]
Comment (rexwangcc) [reply Dockerfile:1]: Yup I tried with `clojure:openjdk-boot` earlier, the main reason is that WFL build requires pulling different repos, some of which are still private. Which means we have to set up git and git tokens for doing so. Not sure how hard it would be on Github Actions but it's defo a pain when building it locally.

[the remark this answers is not in the document]
Comment (rexwangcc) [reply Dockerfile:1]: @tbl3rd I'm wondering how feasible and stable `2.ii` is, that seems more appealing to me.

File: src/zero/zero.clj [hunk 2]
```diff
@@ -58,3 +58,6 @@
               str/trim
               edn/read-string)
       {:version "SOME BOGUS VERSION"}))
+
+
+(get-the-version)
```
Comment (samanehsan) [inline src/zero/zero.clj:63]: Why is this added here?

Comment (tbl3rd) [reply src/zero/zero.clj:63]: Looks like debugging cruft.

Comment (rexwangcc) [reply src/zero/zero.clj:63]: Yeah, that should be removed from here... will do

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (tbl3rd) [reply Dockerfile:1]: The JDK shouldn't matter. OK to use the latest.
